src/core/AudioEngine.cpp

```diff
@@ -134,15 +133,11 @@ AudioEngine::AudioEngine( bool renderOnly ) :
 	// allocte the FIFO from the determined size
 	m_fifo = new Fifo( fifoSize );
 
-	// now that framesPerPeriod is fixed initialize global BufferManager
-	BufferManager::init( m_framesPerPeriod );
+	m_outputBufferRead = new std::array<surroundSampleFrame, m_framesPerPeriod>;
+	m_outputBufferWrite = new std::array<surroundSampleFrame, m_framesPerPeriod>;
```

Comment:
I would use `std::make_unique<surroundSampleFrame[]>` here instead and change those variables to be of type `std::unique_ptr<surroundSampleFrame[]>`.
```suggestion
	m_outputBufferRead = std::make_unique<surroundSampleFrame[]>(m_framesPerPeriod);
	m_outputBufferWrite = std::make_unique<surroundSampleFrame[]>(m_framesPerPeriod);
```

Reply:
Okk i see. Btw if i were to use unique pointers, would i need to modify all functions using the arrays? Or is there something similar to vector->data() which i can use to expose the underlying array?

Reply:
> Okk i see. Btw if i were to use unique pointers, would i need to modify all functions using the arrays? Or is there something similar to vector->data() which i can use to expose the underlying array?

To expose the underlying array of a `std::unique_ptr`, you can use `std::unique_ptr::get`. Check [cppreference](https://en.cppreference.com/w/cpp/memory/unique_ptr) for more info. It's a reference for the C++ programming language that is very useful to know about when you need specific details on something.